plugin/audit_log/filter.c

```diff
@@ -0,0 +1,242 @@
+/* Copyright (c) 2016 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#include <my_global.h>
+#include <my_sys.h>
+#include <my_user.h>
+#include <mysql_com.h>
+#include <hash.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "filter.h"
+
+typedef struct
+{
+  char name[USERNAME_LENGTH + HOSTNAME_LENGTH];
+  size_t length;
+} account;
+
+static HASH include_accounts;
+static HASH exclude_accounts;
+
+#if defined(HAVE_PSI_INTERFACE)
+
+static PSI_rwlock_key key_LOCK_account_list;
+static PSI_rwlock_info all_rwlock_list[]=
+{{ &key_LOCK_account_list, "audit_log_filter::account_list", PSI_FLAG_GLOBAL}};
+
+#endif
+
+mysql_rwlock_t LOCK_account_list;
+
+/*
+  Initialize account
+*/
+static
+void account_init(account *acc, const char *user, size_t user_length,
+                                const char *host, size_t host_length)
+{
+  DBUG_ASSERT(user_length + host_length + 2 <= sizeof(acc->name));
+
+  memcpy(acc->name, user, user_length);
+  memcpy(acc->name + user_length + 1, host, host_length);
+  acc->name[user_length]= '@';
+  acc->name[user_length + host_length + 1]= 0;
+  acc->length= user_length + host_length + 1;
+}
+
+/*
+  Allocate memory and initialize new account
+*/
+
+static
+account *account_create(const char *user, size_t user_length,
+                        const char *host, size_t host_length)
+{
+  account *acc= (account *) my_malloc(sizeof(account),
+                                      MYF(MY_FAE | MY_ZEROFILL));
+
+  account_init(acc, user, user_length, host, host_length);
+
+  DBUG_ASSERT(user_length + host_length + 2 <= sizeof(acc->name));
+
+  memcpy(acc->name, user, user_length);
+  memcpy(acc->name + user_length + 1, host, host_length);
+  acc->name[user_length]= '@';
+  acc->name[user_length + host_length + 1]= 0;
```

Comment:
Yup. account_init was made of account_create and I just forgot to remove the rest...